crates/router/src/core/errors.rs

```diff
@@ -241,6 +241,8 @@ pub enum ConnectorError {
     ResponseHandlingFailed,
     #[error("Missing required field: {field_name}")]
     MissingRequiredField { field_name: &'static str },
+    #[error("Missing required fields: [field_names]")]
```

Comment:
This would end up returning an error message which literally contains `Missing required fields: [field_names]`, without the substitution. To fix it, we'd have to enclose it within `{}`, similar to how we do it for Rust's format strings.

```suggestion
    #[error("Missing required fields: {field_names:?}")]
```

(Removing the square brackets since the `Debug` implementation of `Vec` includes the enclosing square brackets.)

Reply:
Okie, here is another crate `thiserror` is used to help with the errors. Update the usage

Reply:
Yes, we use the `thiserror` crate for helping us define error types as enums with user/developer-friendly error messages.